### PR TITLE
refactor: use `ya.sync` to simplify the code

### PR DESCRIPTION
**Pull request by sxyazi**
Needs the latest main branch of Yazi at the moment.

### CODE REVIEW

**Comment by dedukun** [comment]
Hi, I reverted the commit because I can't get it working on my end.
I installed the git version from cargo, so I'm in

```sh
❯ yazi --version
yazi 0.2.3 (dc1718f 2024-02-09)
```

**Comment by sxyazi** [comment]
That's so weird, have you changed the `plugin` command by removing the `--sync` option?

**Comment by dedukun** [comment]
Yeah, however the issue now is about the `folder` used in `save_bookmarks`. It's `nil` for some reason.

File: init.lua [hunk 1]
```diff
@@ -14,41 +14,20 @@ local SUPPORTED_KEYS = {
...
-
-	local key = SUPPORTED_KEYS[idx].on
-
+local save_bookmark = ya.sync(function(idx)
 	state.bookmarks = state.bookmarks or {}
-	state.bookmarks[key] = { cursor = folder.cursor, path = tostring(folder.cwd) }
-end
+	state.bookmarks[#state.bookmarks + 1] = {
+		on = SUPPORTED_KEYS[idx].on,
+		desc = tostring(folder.cwd),
```
Comment (dedukun) [inline init.lua:21]: If I comment out this line the changes work for me, so there's no problem with the `state`, I was probably doing something wrong yesterday.
The issue on my end is just that `folder` is `nil`

Comment (sxyazi) [reply init.lua:21]: Hi, I made a new PR https://github.com/dedukun/bookmarks.yazi/pull/3, it should fix the `folder`

Comment (dedukun) [reply init.lua:21]: Is that `folder` a global variable? Or should it just be `Folder:by_kind(Folder.CURRENT).cwd`?

Comment (sxyazi) [reply init.lua:21]: No, that's a typo. It should be: `local folder = Folder:by_kind(Folder.CURRENT)`

**Pull request merged**